scalafmt-core/shared/src/main/scala/org/scalafmt/util/LiteralOps.scala

```diff
@@ -0,0 +1,70 @@
+package org.scalafmt.util
+
+import org.scalafmt.config.{Case, ScalafmtConfig}
+
+object LiteralOps {
+
+  /**
+    * Prints integer literals with specified case
+    *
+    * Divides literal into three parts:
+    *
+    * 0xFF123L
+    * 0x is a hex prefix
+    * FF123 is a body
+    * L is a long suffix
+    *
+    * literals.long applies to suffix and literals.hex applies to body
+    * */
+  def prettyPrintInteger(
+      str: String
+  )(implicit style: ScalafmtConfig): String = {
+    val sb = new StringBuilder()
+    val body =
+      if (str.startsWith("0x") || str.startsWith("0X")) {
+        sb.append("0x")
```

Comment:
what if someone asks to use 0X instead?

Reply:
This is a place that I doubt. Lowercased `0x` was hardcoded and I keep it unchanged. But may be reasonable to make it obey `literals.hex` setting. What do you think?